email.go

```diff
@@ -449,20 +447,134 @@ func (e *Email) parseSender() (string, error) {
 	}
 }
 
-// SendWithTLS sends an email with an optional TLS config.
-// This is helpful if you need to connect to a host that is used an untrusted
-// certificate.
-func (e *Email) SendWithTLS(addr string, a smtp.Auth, t *tls.Config) error {
+// Returns slice of To, Cc, and Bcc fields, each parsed with mail.ParseAddress()
+func (e *Email) ParsedAddresses() ([]string, error) {
+
 	// Merge the To, Cc, and Bcc fields
 	to := make([]string, 0, len(e.To)+len(e.Cc)+len(e.Bcc))
 	to = append(append(append(to, e.To...), e.Cc...), e.Bcc...)
 	for i := 0; i < len(to); i++ {
 		addr, err := mail.ParseAddress(to[i])
 		if err != nil {
-			return err
+			return []string{}, err
 		}
 		to[i] = addr.Address
 	}
+
+	return to, nil
+}
+
+/*
+SendWithConn sends an email with an existing connection.
+This is helpful if you need to customize connection parameters (timeouts and such),
+or use a custom Conn interface.
+
+	pSTARTTLSCfg provided: Performs STARTTLS negotiation
+	   pSTARTTLSCfg = nil: Skips STARTTLS negotiation
+		 	                 (for already-encrypted connections, or unencrypted ones)
+
+Typical STARTTLS upgrade:
+
+   HOSTNAME := "mailserver.com"
+   AUTH     := smtp.PlainAuth("", Username, Password, HOSTNAME)
+   pTLSCfg  := &tls.Config{ ... }
+
+   // NOTE: Returns an *unencrypted* net.Conn
+   pConn, _ := net.Dial("tcp4", HOSTNAME + ":587")
+
+   // Provide a *tls.Config to upgrade the connection via STARTTLS negotiation
+   pMsg.Email.SendWithConn(pConn, AUTH, HOSTNAME, pTLSCfg)
+
+Or, provide a nil *tls.Config to send unencrypted:
+
+   pMsg.Email.SendWithConn(pConn, AUTH, HOSTNAME, nil)
+
+Or, begin with a TLS/SSL connection in the first place:
+
+   // NOTE: Returns an *encrypted* tls.Conn
+   pConn, _ := tls.Dial("tcp4", HOSTNAME + ":465", pTLSCfg)
+
+   // No need to provied *tls.Config to upgrade the connection, since it's already encrypted
+   pMsg.Email.SendWithConn(pConn, AUTH, HOSTNAME, nil)
+
+*/
+func (e *Email) SendWithConn(
+	iConn         net.Conn,
+	iAuth         smtp.Auth,
+	serverName    string,
+	pSTARTTLSCfg  *tls.Config,
```

Comment:
Small nit - can we rename these parameters to be simpler. E.g. `conn`, `auth`, `serverName`, and `t` (to match `SendWithTLS`)

Reply:
Not a Charles Simonyi fan I take it?

As for the renaming, go for it!

Reply:
Are you waiting on me to do the rename, or is there a button I need to push to put this back in your court?  Not really a GitHub person here...